### DISCUSSION TITLE
How do I configure Marskman as an LSP server for Markdown?

### DISCUSSION THREAD

**Discussion by g-gundam** [lem-project/lem#1531] [Q&A]
I haven't been able to get [Marksman](https://github.com/artempyanykh/marksman) to work as an LSP server for Markdown files, and I'm wondering if someone could help me diagnose the problem.  This is the configuration I currently have.

```lisp
(lem-lsp-mode/lsp-mode::define-language-spec
    (markdown-spec lem-markdown-mode:markdown-mode)
  :language-id "markdown"
  :root-uri-patterns '()
  :command '("marksman" "server")
  :install-command "true"
  :readme-url "https://github.com/artempyanykh/marksman"
  :connection-mode :stdio)
```

It's based on the TypeScript example I found in the [Lem docs](https://lem-project.github.io/usage/LSP/).  Unfortunately, when I open a Markdown file, I see Lem trying to initialize the LSP server, but it seems stuck in that phase.  The editor is responsive, but the LSP server initialization never finishes.

I'm sure Marksman itself is fine, because it works in Helix and Emacs on my system.

**Comment by cxxxr** [comment]
https://github.com/lem-project/lem/pull/1533
This may be a bug.
Could you try again?

**Reply by g-gundam** [reply]
@cxxxr I pulled down the latest changes and recompiled lem.  However, when I tried to load a markdown file in Lem, it was still stuck during the initialization phase.  Did this change work for you?
